### PR TITLE
Add Chain Signatures POC for Cosmos-based chains

### PR DESCRIPTION
**Pull request by ducphamle2**
- Add Chain Signatures POC for Oraichain using `secp256k1` public key type (normal Cosmos-based chains)
- Add Chain Signatures POC for Oraichain using `ethsecp256k1` public key type (Ethermint-based chains)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,8 +36,8 @@ NEAR_ACCOUNT_ID="[NEAR_TESTNET_ACCOUNT]"
 NEAR_PRIVATE_KEY="[NEAR_ACCOUNT_PRIVATE_KEY]"
 MPC_PATH="[MPC_PATH]"
 MPC_CHAIN="[ethereum|bitcoin]"
-MPC_CONTRACT_ID="v1.signer-dev.testnet"
-MPC_PUBLIC_KEY="secp256k1:54hU5wcCmVUPFWLDALXMh1fFToZsVXrx9BbTbHzSfQq1Kd1rJZi52iPa4QQxo6s5TgjWqgpY8HamYuUDzG6fAaUq"
+MPC_CONTRACT_ID="multichain-testnet-2.testnet"
```
Comment (mattlockyer) [inline README.md]: I don't think this is the latest contract.

We're currently using `v1.signer-dev.testnet` AFAIK.

File: README.md [hunk 2]
```diff
@@ -65,35 +65,35 @@ TATUM_API_KEY=""
...
--   -ea - ethereum addressm (EVM)
--   -ba - bitcoin testnet address
--   -da - dogecoin testnet address
--   -ra - ripple testnet address
--   -s - sign sample payload using NEAR account
--   -etx - send ETH
--   -btx - send BTC
--   -dtx - send DOGE (requires API KEY)
--   -rtx - send XRP
+- -ea - ethereum addressm (EVM)
```
Comment (mattlockyer) [inline README.md]: Please explain formatting change, thanks!

File: src/app.ts [hunk 3]
```diff
@@ -1,181 +1,217 @@
...
+
+  // contract deployment and interaction
+
+  // default: deploys nft contract
+  if (edc) {
+    const { address } = await genAddress("ethereum");
+    await ethereum.deployContract({ from: address, path });
+  }
+
+  if (otx) {
```
Comment (mattlockyer) [inline src/app.ts]: can we move this up with the other send methods?

also perhaps encapsulate try, catch logic in a `send` method in `cosmos.ts` like the other modules. Might make `app.ts` cleaner.

**Review by mattlockyer — CHANGES_REQUESTED** [review]
Great PR thank you! I think we just have to iron out a few style things, feel free to add a prettier file etc... if you have time.

File: README.md (hunk 2 above)
Comment (ducphamle2) [reply README.md]: most likely this was due to my local vscode configuration

File: src/app.ts (hunk 3 above)
Comment (ducphamle2) [reply src/app.ts]: good suggestion. I just pushed a new commit addressing your suggestions @mattlockyer

**Comment by mattlockyer** [comment]
I'll try to test this branch locally today. @ducphamle2 do you have any instructions for the README?

- .env vars if needed?
- faucet for testnet tokens?

**Comment by ducphamle2** [comment]
> I'll try to test this branch locally today. @ducphamle2 do you have any instructions for the README?
> 
> * .env vars if needed?
> * faucet for testnet tokens?

The .env var is unchanged. Meanwhile, we do need a faucet. As of now, there's no faucet for the Oraichain network yet, so it's likely that I'll have to send you some for testing

**Review by mattlockyer — APPROVED** [review]

**Pull request merged**